// ng2-materialize/dist/collapsible/collapsible.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var collapsible_item_component_1 = require("./collapsible-item/collapsible-item.component");
var MzCollapsibleComponent = (function () {
    function MzCollapsibleComponent(changeDetectorRef, renderer) {
        this.changeDetectorRef = changeDetectorRef;
        this.renderer = renderer;
    }
    MzCollapsibleComponent.prototype.ngAfterViewInit = function () {
        this.handleDataCollapsible();
        this.initCollapsible();
    };
    MzCollapsibleComponent.prototype.initCollapsible = function () {
        var _this = this;
        var options = {
            accordion: false,
            onClose: this.onClose,
            onOpen: this.onOpen,
        };
        // need setTimeout otherwise loading directly on the page cause an error
        setTimeout(function () { return _this.renderer.invokeElementMethod($(_this.collapsible.nativeElement), 'collapsible', [options]); });
        // forcing changes detection for unit test
        this.changeDetectorRef.detectChanges();
    };
    MzCollapsibleComponent.prototype.handleDataCollapsible = function () {
        if (this.mode) {
            this.renderer.setElementAttribute(this.collapsible.nativeElement, 'data-collapsible', this.mode);
        }
    };
    return MzCollapsibleComponent;
}());
MzCollapsibleComponent.decorators = [
    { type: core_1.Component, args: [{
                selector: 'mz-collapsible',
                template: "<ul #collapsible class=\"collapsible\" [class.popout]=\"popout\"><ng-content select=\"mz-collapsible-item\"></ng-content></ul>",
                styles: ["/deep/ .collapsible-header{clear:both}/deep/ .collapsible-body{clear:both}"],
            },] },
];
/** @nocollapse */
MzCollapsibleComponent.ctorParameters = function () { return [
    { type: core_1.ChangeDetectorRef, },
    { type: core_1.Renderer, },
]; };
MzCollapsibleComponent.propDecorators = {
    'mode': [{ type: core_1.Input },],
    'onClose': [{ type: core_1.Input },],
    'onOpen': [{ type: core_1.Input },],
    'popout': [{ type: core_1.Input },],
    'collapsible': [{ type: core_1.ViewChild, args: ['collapsible',] },],
    'items': [{ type: core_1.ContentChildren, args: [collapsible_item_component_1.MzCollapsibleItemComponent,] },],
};
exports.MzCollapsibleComponent = MzCollapsibleComponent;
